aip/general/0148.md

```diff
@@ -0,0 +1,117 @@
+---
+id: 148
+state: reviewing
+created: 2020-10-06
+placement:
+  category: fields
+  order: 90
+---
+
+# Standard fields
+
+Certain concepts are common throughout any corpus of APIs. In these situations,
+it is useful to have a standard field name that is used consistently to
+communicate that concept.
+
+## Guidance
+
+Standard fields **should** be used to describe their corresponding concept, and
+**should not** be used for any other purpose.
+
+### Resource names and IDs
+
+#### name
+
+Every resource **must** have a `string name` field, used for the resource name
+(AIP-122), which **should** be the first field in the resource.
+
+**Note:** The `_name` suffix **should not** be used to describe other types of
+names unless otherwise covered in this AIP.
+
+#### parent
+
+The `string parent` field refers to the resource name of the parent of a
+collection, and **should** be used in most `List` (AIP-132) and `Create`
+(AIP-133) requests.
+
+#### uid
+
+The `string uid` field refers to a system-assigned, unique identifier for a
+resource. When provided, this field **should** be a [UUID4][].
+[Declarative-friendly resources][] **should** include this field.
+
+<!-- prettier-ignore -->
+[uuid4]: https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)
+
+### Other names
+
+#### display_name
+
+The `string display_name` field **must** be a mutable, user-settable field
+where the user can provide a human-readable name to be used in user interfaces.
+[Declarative-friendly resources][] **should** include this field.
+
+Display names **should not** have uniqueness requirements, and **should** be
+limited to <= 63 characters.
+
+#### title
+
+The `string title` field **should** be the official name of an entity, such as
+a company's name. This is a more formal variant of `string display_name`.
+
+#### given_name
+
+The `string given_name` field **must** refer to a human or animal's given name.
```

Comment:
Do we want to give any guidance on "middle" names? Would `repeated string given_names` be acceptable for that case? (Naming naming is hard...)

Reply:
I just checked our internal protos for HR, and they use both `middle_name` and `maiden_name`. I am not convinced that either is consistently accurate For example, in cultures where our "last name" comes first, I expect it is actually "family given middle", not "family middle given". Similarly, I think `birth_family_name` is likely more consistently accurate than `maiden_name`.

I propose we discuss this question in a follow-up since it is quasi-unrelated to the main purpose of this PR.